nautobot/extras/api/views.py

```diff
@@ -297,6 +295,169 @@ class ApprovalWorkflowStageViewSet(NautobotModelViewSet):
     serializer_class = serializers.ApprovalWorkflowStageSerializer
     filterset_class = filters.ApprovalWorkflowStageFilterSet
 
+    def _validate_stage(self, stage):
+        """Checks if a workflow has an active stage."""
+        return stage.state == ApprovalWorkflowStateChoices.PENDING
+
+    def _has_change_permission(self, user, workflow):
+        """Checks whether the user has 'change' permission on the model under review."""
+        ct = workflow.object_under_review_content_type
+        model_name = ct.model
+        app_label = ct.app_label
+        change_perm = f"{app_label}.change_{model_name}"
+        return user.has_perm(change_perm)
+
+    def _is_user_approver(self, user, stage):
+        """Checks if the user belongs to the group allowed to approve the current stage."""
+        approver_group = stage.approval_workflow_stage_definition.approver_group
+        return approver_group in user.groups.all()
```

Comment:
Can we more efficiently check something like `user in approver_group.users.all()`?